src/components/form/LinkToPotentialNode.tsx

```diff
@@ -5,22 +5,44 @@ import type { LinkProps } from 'react-router-dom';
 import { SearchParams } from 'src/hooks/useNavigatePage';
 import { useResolvedNode } from 'src/utils/layout/NodesContext';
 
+type Props = LinkProps & { children?: React.ReactNode };
+
 /**
  * This component is used to navigate to a potential node. If the node it is supposed
- * to navigate to does not exist, the link will not replace the current page with the
- * page it is supposed to navigate to.
+ * to navigate to does not exist or the node is hidden, the link will render as pure
+ * text instead of a link.
  * @param props
  * @constructor
  */
-export const LinkToPotentialNode = (props: LinkProps) => {
+export const LinkToPotentialNode = (props: Props) => {
   const [_url, queryParams] = props.to.toString().split('?') ?? [];
+
   const url = new URLSearchParams(queryParams);
-  const resolvedNode = useResolvedNode(url.get(SearchParams.FocusComponentId));
+  const componentId = url.get(SearchParams.FocusComponentId);
+  const resolvedNode = useResolvedNode(componentId);
+
   const nodeExists = resolvedNode != null;
-  return (
-    <Link
-      {...props}
-      replace={!nodeExists}
-    ></Link>
-  );
+  const isNodeHidden = resolvedNode?.isHidden();
+  const shouldShowLink = nodeExists && !isNodeHidden;
+
+  if (shouldShowLink) {
+    return (
+      <Link
+        {...props}
+        replace={!nodeExists}
+      ></Link>
+    );
+  }
```

Comment:
In this part `replace={!nodeExists}`, `!nodeExists` should always evaluate to `false`, since we first check that `nodeExists` is true. It looks like it was just left over from the previous code, but could probably be removed?

Reply:
Yes, you're correct. It can be removed. I can do it later today.